### DISCUSSION TITLE
[Dumb Question] How To Create a Custom Module with Doobda?

### DISCUSSION THREAD

**Discussion by Tajcore** [Tecnativa/doodba#562] [General]
Very dumb question but I'm trying to create a custom module ( a todo app ), using this template but after creating the module in [project_name]/odoo/custom/src/private, how does it get reflected in Odoo's app directory?

**Comment by ap-wtioit** [comment]
Afaik when you create a todo app in `[project_name]`/odoo/custom/src/private/`todo_app` the todo app is symlinked into the odoo addons directory in the container /opt/odoo/auto/addons/`todo_app` and the auto addons dir is added to odoo's config. In dev you also should see the `todo_app` in `[project_name]`/odoo/auto/addons/`todo_app` as a symlink. We always work with repos and adding your addon repos the same way as you would any OCA repo to repos.yaml so i haven't used that feature in a while.

Unfortunately the documentation is pretty minimal for [this](https://github.com/Tecnativa/doodba#optodoocustomsrcprivate) but you the tests have a [private directory with addons](https://github.com/Tecnativa/doodba/tree/master/tests/scaffoldings/dotd/custom/src/private).

**Comment by pedrobaeza** [comment]
You have to restart Odoo instance, and update module list for it to appear on your Odoo instance to be installed.

**Reply by pleirb** [reply]
I'm in production mode, check with `ls *.yml -la`, upload two modules to the privete folder, did `docker-compose down` and a `docker-compose up -d` and refresh the list of apps and still can't find the loaded modules. I also ran `docker-compose restart odoo`. What could be happening?

**Comment by pleirb** [comment]
I was able to fix it with the command `docker-compose build --no-cache`

> I'm in production mode, check with `ls *.yml -la`, upload two modules to the privete folder, did `docker-compose down` and a `docker-compose up -d` and refresh the list of apps and still can't find the loaded modules. I also ran `docker-compose restart odoo`. What could be happening?

**Comment by mycroftsnm** [comment]
Hi, im having issues to load a custom private module too, do i need to edit  `addons.yml ` ?
 i added a symlink inside the `private` folder to another folder where i store my modules but they dont appear in odoo 'apps', i already tried re-building the docker img and updating the modules list from the webgui but still not showing

**Reply by ap-wtioit** [reply]
A symlink is just a special text file, the content of the file doesn't change when it's running inside docker. So if you want to use symlinks the paths they are pointing to need to be available and the same inside and outside of docker.

Or you could just develop your plugins inside of the private folder or you can use custom repos.yaml to have all your plugins in one repo.

**Reply by mycroftsnm** [reply]
Oh i see, makes sense, thanks. Im using custom repos too (my  github), and they work nice, but im starting to develop modules together with a friend so we wanted to have a way to test locally before pushing to our repo.

Gonna try moving the addons directly into the private folder 

btw, do you have any recommended workflow to achieve what i want? We have a little server running doodba with our github repo loaded,but we want to be able to develop different modules each one without the need of pushing everytime we need to test something, so i thought that having a local doodba each one with the private folder approach would suit well. 
But im not really sure if this is the correct way to do things. Sorry for the long text

**Reply by ap-wtioit** [reply]
In devel mode doodba should not aggregate the repos (from github) but use the contents of your `odoo/custom/src/repo_name` without the need to commit and push before. See https://github.com/Tecnativa/doodba-copier-template/blob/main/docs/daily-usage.md#development

**Reply by mycroftsnm** [reply]
Perfect! just tried and is exactly what i wanted, thanks again
